diff -ru Archive/SDL-1.2.7/config.sub src/SDL-1.2.7/config.sub
--- Archive/SDL-1.2.7/config.sub	2004-02-18 17:21:55.000000000 +0000
+++ src/SDL-1.2.7/config.sub	2006-07-28 01:45:35.735006000 +0100
@@ -232,6 +232,7 @@
 	| clipper \
 	| d10v | d30v | dlx | dsp16xx \
 	| fr30 | frv \
+	| xbox \
 	| h8300 | h8500 | hppa | hppa1.[01] | hppa2.0 | hppa2.0[nw] | hppa64 \
 	| i370 | i860 | i960 | ia64 \
 	| ip2k \
@@ -991,6 +992,10 @@
 		basic_machine=hppa1.1-winbond
 		os=-proelf
 		;;
+	xbox)
+		basic_machine=i386-pc
+		os=-xbox
+		;;
 	xps | xps100)
 		basic_machine=xps100-honeywell
 		;;
@@ -1234,6 +1239,9 @@
 	-xenix)
 		os=-xenix
 		;;
+	-xbox)
+		os=-xbox
+		;;
 	-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 		os=-mint
 		;;
diff -ru Archive/SDL-1.2.7/configure.in src/SDL-1.2.7/configure.in
--- Archive/SDL-1.2.7/configure.in	2004-02-22 21:31:47.000000000 +0000
+++ src/SDL-1.2.7/configure.in	2006-07-28 01:45:35.738006000 +0100
@@ -2571,6 +2571,33 @@
 
         SYSTEM_LIBS="$SYSTEM_LIBS -ljpeg -ltiff -lpng -lz"
         ;;
+    *-*-xbox)
+        ARCH=win32
+        ac_default_prefix=$prefix/bin
+        AUDIO_SUBDIRS="$AUDIO_SUBDIRS xbox"
+        AUDIO_DRIVERS="$AUDIO_DRIVERS xbox/libaudio_xbox.la"
+        JOYSTICK_SUBDIRS="$JOYSTICK_SUBDIRS xbox"
+        JOYSTICK_DRIVERS="$JOYSTICK_DRIVERS xbox/libjoystick_xbox.la"
+        COPY_ARCH_SRC(src/thread, xbox, SDL_systhread.c)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_systhread_c.h)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_sysmutex.c)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_sysmutex_c.h)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_syssem.c)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_syssem_c.h)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_syscond.c)
+        COPY_ARCH_SRC(src/thread, xbox, SDL_syscond_c.h)
+        COPY_ARCH_SRC(src/timer, xbox, SDL_systimer.c)
+        VIDEO_SUBDIRS="$VIDEO_SUBDIRS xbox"
+        VIDEO_DRIVERS="$VIDEO_DRIVERS xbox/libvideo_xbox.la"
+
+        # The xbox platform requires special setup
+        CFLAGS="$CFLAGS -I$prefix/include -DENABLE_XBOX -DDISABLE_CDROM"
+        case "$build" in
+            *-*-cygwin*)
+                CFLAGS="$CFLAGS -fomit-frame-pointer -std=gnu99 -ffreestanding -nostdlib -fno-builtin -fno-exceptions -mno-cygwin -march=i386"
+                ;;
+        esac
+        ;;
     *)
         AC_MSG_ERROR(Unsupported target:  Please add to configure.in)
         ;;
@@ -2746,6 +2773,7 @@
 src/audio/ums/Makefile
 src/audio/windib/Makefile
 src/audio/windx5/Makefile
+src/audio/xbox/Makefile
 src/video/Makefile
 src/video/aalib/Makefile
 src/video/ataricommon/Makefile
@@ -2782,6 +2810,7 @@
 src/video/XFree86/Xv/Makefile
 src/video/XFree86/Xxf86dga/Makefile
 src/video/XFree86/Xxf86vm/Makefile
+src/video/xbox/Makefile
 src/events/Makefile
 src/joystick/Makefile
 src/joystick/amigaos/Makefile
@@ -2795,6 +2824,7 @@
 src/joystick/mint/Makefile
 src/joystick/riscos/Makefile
 src/joystick/win32/Makefile
+src/joystick/xbox/Makefile
 src/cdrom/Makefile
 src/cdrom/aix/Makefile
 src/cdrom/beos/Makefile
diff -ru Archive/SDL-1.2.7/include/SDL_syswm.h src/SDL-1.2.7/include/SDL_syswm.h
--- Archive/SDL-1.2.7/include/SDL_syswm.h	2004-02-18 17:21:58.000000000 +0000
+++ src/SDL-1.2.7/include/SDL_syswm.h	2006-07-28 01:48:12.927109000 +0100
@@ -110,6 +110,7 @@
 	GR_WINDOW_ID window ;	/* The display window */
 } SDL_SysWMinfo;
 
+#ifndef ENABLE_XBOX
 #elif defined(WIN32)
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
@@ -130,6 +131,8 @@
 	HGLRC hglrc;			/* The OpenGL context, if any */
 } SDL_SysWMinfo;
 
+#endif // ENABLE_XBOX
+
 #elif defined(__riscos__)
 
 /* RISC OS custom event structure */
diff -ru Archive/SDL-1.2.7/src/audio/Makefile.am src/SDL-1.2.7/src/audio/Makefile.am
--- Archive/SDL-1.2.7/src/audio/Makefile.am	2003-08-30 20:13:00.000000000 +0100
+++ src/SDL-1.2.7/src/audio/Makefile.am	2006-07-28 01:45:35.740005000 +0100
@@ -26,7 +26,8 @@
 	sun	\
 	ums	\
 	windib	\
-	windx5
+	windx5 \
+	xbox
 
 DRIVERS = @AUDIO_DRIVERS@
 
diff -ru Archive/SDL-1.2.7/src/audio/SDL_audio.c src/SDL-1.2.7/src/audio/SDL_audio.c
--- Archive/SDL-1.2.7/src/audio/SDL_audio.c	2004-02-18 17:21:58.000000000 +0000
+++ src/SDL-1.2.7/src/audio/SDL_audio.c	2006-07-28 01:45:35.738006000 +0100
@@ -105,6 +105,9 @@
 #ifdef DRENDERER_SUPPORT
 	&DRENDERER_bootstrap,
 #endif
+#ifdef ENABLE_XBOX
+	&XBOX_AudioBootstrap,
+#endif
 	NULL
 };
 SDL_AudioDevice *current_audio = NULL;
diff -ru Archive/SDL-1.2.7/src/audio/SDL_sysaudio.h src/SDL-1.2.7/src/audio/SDL_sysaudio.h
--- Archive/SDL-1.2.7/src/audio/SDL_sysaudio.h	2004-02-18 17:21:58.000000000 +0000
+++ src/SDL-1.2.7/src/audio/SDL_sysaudio.h	2006-07-28 01:45:35.739005000 +0100
@@ -166,6 +166,9 @@
 #ifdef DRENDERER_SUPPORT
 extern AudioBootStrap DRENDERER_bootstrap;
 #endif
+#ifdef ENABLE_XBOX
+extern AudioBootStrap XBOX_AudioBootstrap;
+#endif
 #ifdef MMEAUDIO_SUPPORT
 extern AudioBootStrap MMEAUDIO_bootstrap;
 #endif
diff -ru Archive/SDL-1.2.7/src/joystick/Makefile.am src/SDL-1.2.7/src/joystick/Makefile.am
--- Archive/SDL-1.2.7/src/joystick/Makefile.am	2003-08-30 20:13:05.000000000 +0100
+++ src/SDL-1.2.7/src/joystick/Makefile.am	2006-07-28 01:45:35.740005000 +0100
@@ -5,7 +5,7 @@
 
 # Define which subdirectories need to be built
 SUBDIRS = @JOYSTICK_SUBDIRS@
-DIST_SUBDIRS = dummy amigaos beos bsd darwin dc linux macos mint riscos win32
+DIST_SUBDIRS = dummy amigaos beos bsd darwin dc linux macos mint riscos win32 xbox
 
 DRIVERS = @JOYSTICK_DRIVERS@
 
diff -ru Archive/SDL-1.2.7/src/thread/Makefile.am src/SDL-1.2.7/src/thread/Makefile.am
--- Archive/SDL-1.2.7/src/thread/Makefile.am	2004-02-18 17:22:02.000000000 +0000
+++ src/SDL-1.2.7/src/thread/Makefile.am	2006-07-28 01:45:35.741005000 +0100
@@ -12,7 +12,8 @@
                $(srcdir)/irix		\
                $(srcdir)/linux		\
                $(srcdir)/pth		\
-               $(srcdir)/win32
+               $(srcdir)/win32 \
+               $(srcdir)/xbox
 
 # Older versions of Linux require an asm clone() implementation
 if USE_CLONE
diff -ru Archive/SDL-1.2.7/src/timer/Makefile.am src/SDL-1.2.7/src/timer/Makefile.am
--- Archive/SDL-1.2.7/src/timer/Makefile.am	2004-02-18 17:22:03.000000000 +0000
+++ src/SDL-1.2.7/src/timer/Makefile.am	2006-07-28 01:45:35.742005000 +0100
@@ -11,7 +11,8 @@
                $(srcdir)/macos \
                $(srcdir)/mint \
                $(srcdir)/riscos \
-               $(srcdir)/win32
+               $(srcdir)/win32 \
+               $(srcdir)/xbox
 
 # Include the architecture-independent sources
 COMMON_SRCS = SDL_timer.c SDL_timer_c.h SDL_systimer.h
diff -ru Archive/SDL-1.2.7/src/video/Makefile.am src/SDL-1.2.7/src/video/Makefile.am
--- Archive/SDL-1.2.7/src/video/Makefile.am	2003-08-30 20:13:06.000000000 +0100
+++ src/SDL-1.2.7/src/video/Makefile.am	2006-07-28 01:45:35.746004000 +0100
@@ -9,7 +9,8 @@
                wincommon windib windx5 \
                maccommon macdsp macrom riscos quartz \
                bwindow ps2gs photon cybergfx epoc picogui \
-               ataricommon xbios gem dc qtopia XFree86
+               ataricommon xbios gem dc qtopia XFree86 \
+               xbox
 
 DRIVERS = @VIDEO_DRIVERS@
 
diff -ru Archive/SDL-1.2.7/src/video/SDL_sysvideo.h src/SDL-1.2.7/src/video/SDL_sysvideo.h
--- Archive/SDL-1.2.7/src/video/SDL_sysvideo.h	2004-02-18 17:22:04.000000000 +0000
+++ src/SDL-1.2.7/src/video/SDL_sysvideo.h	2006-07-28 01:47:29.162762000 +0100
@@ -38,7 +38,9 @@
  */
 
 /* OpenGL is pretty much available on all Windows systems */
-#ifdef WIN32
+#ifdef ENABLE_XBOX
+#include <openxdk/openxdk.h>
+#elif define WIN32
 #ifndef _WIN32_WCE
 #define HAVE_OPENGL
 #endif
@@ -414,6 +416,9 @@
 #ifdef ENABLE_RISCOS
 extern VideoBootStrap RISCOS_bootstrap;
 #endif
+#ifdef ENABLE_XBOX
+extern VideoBootStrap XBOX_bootstrap;
+#endif
 /* This is the current video device */
 extern SDL_VideoDevice *current_video;
 
diff -ru Archive/SDL-1.2.7/src/video/SDL_video.c src/SDL-1.2.7/src/video/SDL_video.c
--- Archive/SDL-1.2.7/src/video/SDL_video.c	2004-02-18 17:22:04.000000000 +0000
+++ src/SDL-1.2.7/src/video/SDL_video.c	2006-07-28 01:45:35.744005000 +0100
@@ -123,6 +123,9 @@
 #ifdef ENABLE_DUMMYVIDEO
 	&DUMMY_bootstrap,
 #endif
+#ifdef ENABLE_XBOX
+	&XBOX_bootstrap,
+#endif
 	NULL
 };
 
